src/pedalintandem.py

```diff
@@ -0,0 +1,212 @@
+import json
+from datetime import datetime, timedelta
+from curl_cffi import requests
+from common.tz import IST
+from bs4 import BeautifulSoup
+import re
+
+BASE_URL = "https://www.pedalintandem.com"
+
+def fetch_events_links(session):
+	res = session.get(f"{BASE_URL}/experiences")
+	soup = BeautifulSoup(res.text, 'html.parser')
+	events = soup.select('div.single-experience')
+
+	event_links = map(lambda x: x.find('a')['href'], events)
+
+	return event_links
+
+def fetch_events(event_links, session):
+	events = []
+	for event_link in event_links:
+
+		event_page = session.get(f"{BASE_URL}{event_link}")
+		event = BeautifulSoup(event_page.text, 'html.parser')
+		location = event.select_one('div.location').get_text().strip()
+
+		date_selector = event.select_one('div.product-variations-varieties select')
+
+		# Fetch data of the events which would be happening in future otherwise skip
+		if ( "disabled" in date_selector.attrs):
+			continue
+		
+		if not ( bool(re.search('bangalore', location, re.IGNORECASE)) or
+			bool(re.search('bengaluru', location, re.IGNORECASE)) or
+			bool(re.search('arkavathi', location, re.IGNORECASE)) or
+			bool(re.search('avalahalli', location, re.IGNORECASE)) or
+			bool(re.search('avathi', location, re.IGNORECASE)) or
+			bool(re.search('devarayanadurga', location, re.IGNORECASE)) or
+			bool(re.search('gunjur', location, re.IGNORECASE)) or
+			bool(re.search('hennur', location, re.IGNORECASE)) or
+			bool(re.search('Hesaraghatta', location, re.IGNORECASE)) or
+			bool(re.search('kanakapura', location, re.IGNORECASE)) or
+			bool(re.search('malleshwaram', location, re.IGNORECASE)) or
+			bool(re.search('indiranagar', location, re.IGNORECASE)) or
+			bool(re.search('manchanabele', location, re.IGNORECASE)) or
+			bool(re.search('pedal', location, re.IGNORECASE)) or
+			bool(re.search('pitstop', location, re.IGNORECASE)) or 
+			bool(re.search('rajankunte', location, re.IGNORECASE)) ):
+			continue
+
+		duration = event.select_one('div.duration').get_text().strip()
+		if bool(re.search('/', duration)):
+			continue
+
+		events.append(event)
+
+	return events
+
+def make_event(event):
+	heading = event.select_one('div.heading').get_text().strip()
+
+	location = event.select_one('div.location').get_text().strip()
+
+	offers = {}
+	addOn = {}
+	offers_selector = event.select_one('div.cart-details')
+	opts = offers_selector.select('div.product-variations select[name="variation_id"] option')
+	for opt in opts:
+		opt_name = opt.get_text()
+		price = opt['data-price-after-discount']
+		price = price.replace("\u20b9", "")
+		if "rent" in opt_name or "transport" in opt_name:
+			addOn[opt_name] = price
+		else:
+			offers[opt_name] = price
+	offers['addOn'] = addOn
```

Comment:
refactor to a separate get_price or something similar.

Reply:
Done.